docs/Reference-for-template.json.md

```diff
@@ -197,6 +197,58 @@ String optional parameter, replaces TargetFrameworkOverride:
 }
 ```
 
+#### Multichoice symbols specifics
+Multichoice symbols have similar behavior and usage scenarios as [C# Flag enums](https://docs.microsoft.com/en-us/dotnet/api/system.flagsattribute) - they express a range of possible values (not a single value - unlike the plain [choice symbol](#choice-sample))
+
+There are some specifics in behavior of multichoice symbols that are worth noting:
+
+* Condition evaluation - closer described in [Conditions document](Conditions.md#multichoice-literals).
+* [`Switch` symbol](Available-Symbols-Generators.md#switch) evaluation - conditions are evaluated by identical evaluator as preprocessing conditions (previous bulet point). 
+* Tab completion on CLI - Tab completion is works identically as for standard choice symbol, user can specify multiple values via repeating the option switch - e.g.: `dotnet new MyTemplate --MyParameter value1 --MyParameter value2`
```

Comment:
we should talk about the semantics of this when we implement it - System.CommandLine also allows for multiple tokens to be supplied as part of one option value (in this case that would look like `dotnet new MyTemplate --MyParameter value1 value 2`) and toggling that behavior is a single line of code.  We should check the ergonomics of other CLIs and develop an opinion on which is better.

Reply:
@baronfel Good point. I wasn't aware that both syntaxes of attaching multiple options to same arguments are supported - it turns out they are: https://docs.microsoft.com/en-us/dotnet/standard/commandline/syntax#argument-arity
So in the spirit of consistency we should make sure we support both (it should be almost for free, only thing I'm not sure about is tab completion - I will look into that). I'll update it in the proposal.